### PR TITLE
Add support for charty

### PR DESCRIPTION
**Pull request by kozo2**

### CODE REVIEW

File: lib/rubydown.rb [hunk 1]
```diff
@@ -29,6 +30,13 @@ def to_html
   end
 end
 
+Charty::RenderContext.prepend Module.new {
+  def render(filename=nil)
+    super
+    File.open(filename)
```
Comment (znz) [inline lib/rubydown.rb]: This raises TypeError when filename is nil.

File: exe/rubydown [hunk 2]
```diff
@@ -62,6 +62,9 @@ class EvalDoc < Kramdown::Document
     case
     when eval_result.respond_to?(:to_html)
       Kramdown::Document.new(eval_result.to_html, input: :html).root
+    when eval_result.instance_of?(File) && eval_result.path[-4,4] == ".png"
+      img_b64 = Base64.encode64(File.binread(eval_result))
```
Comment (znz) [inline exe/rubydown]: `File.binread` accepts file path string.
If `eval_result` is IO, use `eval_result.read` after open with `'rb'` (read binary mode).
Or pass `eval_result` as file path string.

File: exe/rubydown [hunk 3]
```diff
@@ -62,6 +62,9 @@ class EvalDoc < Kramdown::Document
     case
     when eval_result.respond_to?(:to_html)
       Kramdown::Document.new(eval_result.to_html, input: :html).root
+    when eval_result.instance_of?(File) && eval_result.path[-4,4] == ".png"
```
Comment (znz) [inline exe/rubydown]: I prefer `path.end_with?('.png')` to `path[-4,4] == ".png"`.

**Review by znz — CHANGES_REQUESTED** [review]

File: exe/rubydown (hunk 3 above)
Comment (kozo2) [reply exe/rubydown]: Thanks, but I will use
```
File.extname(eval_result.path) == ".png"
```
instead.

**Pull request merged**